Samples/runtime/simulator-with-overrides/FaultyMeasurementsSimulator.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+using Microsoft.Quantum.Simulation.Simulators;
+using Microsoft.Quantum.Simulation.Core;
+using Microsoft.Quantum.Intrinsic;
+
+namespace Microsoft.Quantum.Samples.SimulatorWithOverrides
+{
+    /// <summary>
+    /// A simulator which extends QuantumSimulator and redefines measurement operation 
+    /// to return the correct measurement result with certain probability 
+    /// and the opposite measurement result in the rest of the cases.
+    /// </summary>
+    public class FaultyMeasurementsSimulator : QuantumSimulator
+    {
+        /// <summary>
+        /// The overriding definition for operation M
+        /// </summary>
+        public class M : QSimM
+        {
+            /// <summary>
+            /// The probability with which the measurement result will be flipped.
+            /// </summary>
+            const double flipProbability = 0.1;
+
+            /// <summary>
+            /// Random numbers generator used to decide when to flip the result.
+            /// </summary>
+            private static readonly System.Random rnd = new System.Random();
+
+            /// <summary>
+            /// The X gate operation used to adjust the state to match flipped measurement results.
+            /// </summary>
+            private static IUnitary<Qubit> gateX;
+
+            public M(FaultyMeasurementsSimulator m) : base(m) { }
```

Comment:
Since you're using `ResetAll` below, you'll likely want to initialize the base `QuantumSimulator` to throw when qubits aren't in the |0⟩ state at release-time. In my experience, faulty measurements cause reset operations to incorrectly reset, such that the simulator will exception out pretty frequently.

Reply:
That's the default behavior of `QuantumSimulator`, I don't think any special initialization is required? (Helped me to find bug in my implementation yesterday, so it's definitely "on")

Reply:
Sorry, typo on my part, I should have said "to _not_ throw."